tiledb/storage_format/uri/parse_uri.cc

```diff
@@ -31,12 +31,30 @@
  */
 
 #include "parse_uri.h"
+
 #include <algorithm>
 #include <cinttypes>  // For sscanf PRId64
+#include <set>
 #include <sstream>
+#include "tiledb/common/common.h"
+#include "tiledb/sm/misc/tdb_time.h"
+#include "tiledb/sm/misc/uuid.h"
 
 namespace tiledb::sm::utils::parse {
 
+std::string generate_fragment_name(
```

Comment:
- `parse_uri.cc` doesn't seem like the right place for this, although perhaps we could simply rename to `fragment_uri.cc`
- This partially duplicates `ArraySchema::generate_uri`. I'd like to avoid that duplication (ArraySchema isn't a great place for it either, admittedly)

Reply:
If we want to use the same function for the fragment URIs and array schema URIs, then `fragment_uri.cc` isn't really the right name either. Not sure what a better name would be though. Also, should I change the namespace? If so, just for this function, or for all of the functions in `parse_uri.cc`?